File: 1127-last-stone-weight/1127-last-stone-weight.py
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        stones.sort(reverse=True)

        while len(stones) > 1:
            stone1 = stones.pop(0)
            stone2 = stones.pop(0)

            new_weight = abs(stone1 - stone2)

            if new_weight == 0: continue

            insert_index = len(stones)
            for i in range(len(stones)):
                if stones[i] < new_weight:
                    insert_index = i
                    break

            stones.insert(insert_index, new_weight)

        if len(stones) == 0: return 0

        return stones[0]

File: 1127-last-stone-weight/1127-last-stone-weight.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
        if(stones.size()==1){
            return stones[0];
        }
        if(stones.size()==0){
            return 0;
        }
        priority_queue<int> pq;

        //vector.sort(stones.begin(),stones.end(),greater<>());
        for( int stone: stones){
            pq.push(stone);
        }

        while(pq.size()>1){
            int one = pq.top();
            pq.pop();
            int two = pq.top();
            pq.pop();
            if(one != two){
                pq.push(one - two);
            }
        }
        return pq.size() == 1 ? pq.top() : 0;
    }
};